### PR TITLE
Bugfix: convert id to html token before usage as html ID

### PR DESCRIPTION
**Pull request by Perry26**
Another nice story to show that you should never thrust tokens from your user input. (The . becomes a class selector when you pass it to d3)

This will make the input files work with our application.

### CODE REVIEW

**Comment by christojeffrey** [comment]
is it possible to handle this is converter instead, to minimize the cost?

File: svelte/src/routes/v2/scripts/draw/index.ts [hunk 1]
```diff
@@ -7,6 +7,10 @@ import type { ConfigInterface, DrawSettingsInterface } from '../../types';
 const SVGSIZE = 800;
 const SVGMARGIN = 50;
 
+function toHTMLToken(string: string) {
+	return string.replace(/[^A-Za-z0-9]/g, '--');
```
Comment (christojeffrey) [inline svelte/src/routes/v2/scripts/draw/index.ts:11]: aah this will change anything that is not alphanumeric, to "--" right?

**Comment by Adityapnn811** [comment]
Oh god, great job on finding the problem

**Comment by Perry26** [comment]
> is it possible to handle this is converter instead, to minimize the cost?

It is, but the consequence would be that either: 
 - The original name (which the user expects to see in the ui) is no longer avaidable anywhere
 - We need to store the name twice, which wastes memory and might cause extra maintanance/confusion down the road. (Whearas if you forget toHTMLToken somewhere now, you'll notice literally as soon as you load new data).

**Comment by christojeffrey** [comment]
> > is it possible to handle this is converter instead, to minimize the cost?
> 
> It is, but the consequence would be that either:
> 
> * The original name (which the user expects to see in the ui) is no longer avaidable anywhere
> * We need to store the name twice, which wastes memory and might cause extra maintanance/confusion down the road. (Whearas if you forget toHTMLToken somewhere now, you'll notice literally as soon as you load new data).

I see. assuming we want to keep the original name, you would take heavier computation rather than adding more data (the second option)?

**Comment by Perry26** [comment]
> Owner
> ###

This feels like a temporary / debug only solution anyway (what if a class name actually contains dashes? Or what if a class name has other weird characters?).
In the future, we probably want DOM nodes to be addres by a uuid or something, but I would not suggest implementing that quite yet for debugging reasons. (Go ahead and do it if you disagree).

Also, I'm not sure if this will be a bottleneck performance-wise. In my experience, the rendering back-end is going to dominate the running time eventually (there is only so much you can do with svg before it starts lagging), though I hope to be proven wrong

**Comment by christojeffrey** [comment]
alright then. though small, I don't think letting go of expensive computation stuffs is a good practice because they will add up. though I am guilty of this too lol

I guess you've considered it and this is what you choose to go with. Thanks for considering it.

I'll approve it now

**Review by christojeffrey — APPROVED** [review]

**Pull request merged**